scripts/deployer.ts

```diff
@@ -0,0 +1,197 @@
+import { ethers } from 'hardhat';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from 'hardhat-deploy/types';
+import 'hardhat-deploy-ethers';
+import hre from 'hardhat';
+import * as fs from 'fs';
+import { ContractReceipt, ContractTransaction } from 'ethers';
+import mainnetData from '../mainnet_deployed.json';
+import testnetData from '../testnet_deployed.json';
+import dummyData from '../example_testnet_deployed.json';
+import { getCurrentTimestamp } from 'hardhat/internal/hardhat-network/provider/utils/getCurrentTimestamp';
+
+let deployedData = dummyData;
+let fileName = 'example_testnet_deployed.json';
+// refactor:
+//  mainnet_deployed.json
+//  testnet_deployed.json
+//  local_deployed.json
+
+// URL
+const mainnetURL = 'https://gnosisscan.io/address/';
+const testnetURL = 'https://goerli.etherscan.io/address/';
+
+// Chain Vendor
+const blockChainVendor = hre.network.name;
+const networkID = hre.network.config.chainId;
+
+// all contract addresses will be set here
+let factory = '';
+let staking = '';
+let postage = '';
+let oracle = '';
+let redistribution = '';
+
+// all block numbers will be set here
+let factoryBlock = 0;
+let stakingBlock = 0;
+let postageBlock = 0;
+let oracleBlock = 0;
+let redistributionBlock = 0;
```

Comment:
This is not really good approach 😅 I would instead suggest passing these values as function parameters instead of having them as "global variables". Simply construct an object that will contain these properties as you deploy them and then pass it into the functions that process them.

Reply:
If the problem is that you do not have all the info at the same time then you can use [`Partial<>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) helper type that allows all the direct properties of the type to be `undefined`.

Reply:
it also looked funny btw :laughing: